api/test/wfl/tools/datasets.clj

```diff
@@ -0,0 +1,45 @@
+(ns wfl.tools.datasets
+  (:require [clojure.string :as str]
+            [clojure.data.json :as json]
+            [wfl.environment :as env]
+            [wfl.service.datarepo :as datarepo]
+            [wfl.service.google.storage :as storage])
+  (:import (java.util UUID)))
+
+(defn unique-dataset-request
```

Comment:
I think it's better that we don't adjust permissions in code like this. Better we require users to grant permissions than we grant them ourselves (likewise with all IAM-related endpoints)
- Add the TDR dev SA to be a reader of this bucket in gotc-deploy
- Figure our how TDR production works and do the same so we don't have to test permissions in production.